src/lib/mdns/ServiceNaming.cpp

```diff
@@ -130,7 +132,10 @@ CHIP_ERROR MakeServiceSubtype(char * buffer, size_t bufferLen, DiscoveryFilter s
         {
             return CHIP_ERROR_INVALID_ARGUMENT;
         }
-        requiredSize = snprintf(buffer, bufferLen, "_D%u", subtype.code);
+        requiredSize = snprintf(buffer, bufferLen, "_D%" PRIu64, subtype.code);
+        break;
+    case DiscoveryFilterType::kCompressedFabricId:
+        requiredSize = snprintf(buffer, bufferLen, "_I%" PRIX64, subtype.code);
```

Comment:
PRIX64/PRIu64 is not available on all platforms IIRC. That is why there is `ChipLogFormatX64` used. I wonder if this will break at runtime on tiny printfs.

A simple way to fix is to:

1- For all types that fit and where you use `PRIu64`, keep the original `%u` and do a `static_cast<unsigned>(subtype.code)` in the snprintf args.
2- For compressedFabricId, use `BytesToHex` for the hex conversion into a temp stack buffer and then use a `%s` in the `snprintf`. That would make the whole thing "low-touch" without worry about small printf libs without 64 bit support.

Reply:
Changed in next commit - added BytesToHex for uint64 with tests.